src/shogun/lib/DynamicArray.h

```diff
@@ -335,14 +390,25 @@ template <class T> class CDynamicArray :public CSGObject
 		 * @param e element to append
 		 */
 		inline void push_back(T e)
-		{ m_array.push_back(e); }
+		{
+			if (num_elements < int32_t(m_array.size()))
+			{
+				m_array[num_elements] = e;
+			}
+			else
+			{
+				m_array.push_back(e);
+			}
+			++num_elements;
+		}
 
 		/** STD VECTOR compatible. Delete array element at the end
 		 *  of array.
 		 */
 		inline void pop_back()
 		{
 			m_array.pop_back();
+			--num_elements;
```

Comment:
i mean i'm not so sure how was this previously done... was the size of the array changed when deleted? if not then of course you need to nullify the element... ```nullptr``` only works if the stored element of DynamicArray is a pointer right? i.e. if it's index_t or other primitive time you have to come up with a different value. but then again `--num_elements` should be enough w/o touching the array itself as you shouldnt be able to read elements past who's index is `index >= num_elements`